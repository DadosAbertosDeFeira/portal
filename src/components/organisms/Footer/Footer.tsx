import { Link } from "atoms/Link";
import React from "react";
import { FiInstagram, FiTwitter } from "react-icons/fi";

export type FooterProps = {};

export function Footer({ ...props }: FooterProps) {
  return (
    <footer
      className="flex flex-col items-center justify-center gap-4 bg-[#DBE6EF] p-5 text-gray-900 md:relative"
      {...props}
    >
      <p className="grow text-center font-medium">
        Â© {new Date().getFullYear()} Dados abertos de Feira. Todos os direitos
        reservados
      </p>
      <nav className="right-8 md:absolute">
        <ul className="flex flex-row flex-nowrap gap-5">
          <li>
            <Link href="https://twitter.com/DadosDeFeira">
              <span className="sr-only">Twitter</span>
              <FiTwitter size={25} />
            </Link>
          </li>
          <li>
            <Link href="https://www.instagram.com/dadosabertosdefeira/">
              <span className="sr-only">Instagram </span>
              <FiInstagram size={25} />
            </Link>
          </li>
        </ul>
      </nav>
    </footer>
  );
}
