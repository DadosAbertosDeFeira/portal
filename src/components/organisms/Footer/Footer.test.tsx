import { render } from "@testing-library/react";

import { Footer, type FooterProps } from "./Footer";

describe("Footer", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  const makeSut = ({ ...props }: Partial<FooterProps>) => {
    return render(<Footer {...props} />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });
});
