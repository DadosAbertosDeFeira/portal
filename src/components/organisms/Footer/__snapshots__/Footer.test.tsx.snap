// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `[Function]`;
